File: src/types/prerequisites/single/CulturePrerequisite.ts
import * as DB from "tsondb/schema/dsl"
import { CultureIdentifier } from "../../_Identifier.js"
import { DisplayOption } from "../DisplayOption.js"

export const CulturePrerequisite = DB.TypeAlias(import.meta.url, {
  name: "CulturePrerequisite",
  comment: "Requires a specific culture or one of a specific set of cultures.",
  type: () =>
    DB.Object({
      id: DB.Required({
        comment: "The culture’s identifier.",
        type: CultureIdentifier(),
      }),
      display_option: DB.Optional({
        type: DB.IncludeIdentifier(DisplayOption),
      }),
    }),
})

File: Sources/OptolithDatabaseSchema/GeneratedTypes/prerequisites/single/CulturePrerequisite.swift
//
//  CulturePrerequisite.swift
//  OptolithDatabaseSchema
//

/// Requires a specific culture or one of a specific set of cultures.
public struct CulturePrerequisite: EntitySubtype {
    /// The culture's identifier.
    public let id: CultureIdentifier

    public let displayOption: DisplayOption?

    public init(id: CultureIdentifier, displayOption: DisplayOption? = nil) {
        self.id = id
        self.displayOption = displayOption
    }

    private enum CodingKeys: String, CodingKey {
        case id = "id"
        case displayOption = "display_option"
    }
}
